File: HW/HW8/longestPalin.java
package HW.HW8;

public class longestPalin {
    
    public int longestPalindrome(String s) {
        // ASCII characters
        int [] count = new int [128];
        for (char c: s.toCharArray())
        {
            count[c]++;
        }
        int result = 0;
        for (int current: count)
        {
            result += current / 2 * 2;
            if (result == 0 && current % 2 == 1)
            {
                result++;
            }
        }
        return result;

    }
}

File: HW/HW8/longestPalin.py
def longestPalindrome(self, s):
    count = [0] * 128
    for c in s:
        count[ord(c)] += 1
    result = 0
    for current in count:
        result += current // 2 * 2
        if result % 2 == 0 and current % 2 == 1:
            result += 1
    return result
